api/src/main/java/org/apache/iceberg/expressions/BoundAggregate.java

```diff
@@ -37,11 +37,34 @@ public BoundReference<?> ref() {
     return term().ref();
   }
 
+  public Types.NestedField nestedField(int index) {
+    if (op() == Operation.COUNT_STAR) {
+      return Types.NestedField.required(index, "COUNT(*)", Types.LongType.get());
+    } else if (op() == Operation.COUNT) {
+      return Types.NestedField.required(
+          index, "COUNT(" + term().ref().name() + ")", Types.LongType.get());
```

Comment:
These strings aren't correct because there are two types of terms. A term can be a reference (e.g., a column named `id`) or a transformed reference (e.g. `bucket(16, id)`).

The `ref()` method is convenient for getting the underlying reference, but this can't just get the ref from the term and assume that the term is a ref.

Reply:
@rdblue Thanks for your comment. 

Seems to me that we can't aggregate on a  transformed reference.
I currently have 
```
public class BoundAggregate<T, C> extends Aggregate<BoundTerm<T>> implements Bound<C>
```
Should I change to the following?
```
public class BoundAggregate<T, C> extends Aggregate<BoundReference<T>> implements Bound<C>
```

Reply:
I would not change the definition. Just catch that case and throw an exception.

Reply:
I was able to get transformed values working in #6405 by calling `term().eval(struct)`.